test/torchscript_bc_test/README.md

```diff
@@ -0,0 +1,23 @@
+# Torchscript Backward Compatibility Test Suite
+
+This directory contains tools to generate Torchscript object of a specific torchaudio version (given that you have the corresponding environments setup correctly) and validate it in the current version.
+
+## Usage
+
+### Generate torchscript object
+
+```
+./main.py --mode generate --version 0.6.0
+```
+
+will generate Torchscript dump files in [`assets`](./assets/) directory. This requries your Python runtime to have torchaudio `0.6.0`.
+
+
+### Validate torchscript object
+
+
+```
+./main.py --mode validate --version 0.6.0
```

Comment:
This is more appropriate in a unittest. Thoughts?

Reply:
I do not think so because unit tests can be just run. This test suite has a set of constraints and is more opinionated in how it should be run. Mixing them would make it harder to manage.
What is your rational on putting this in a unit test?

Reply:
I was saying this because the validation takes a checkpoint and loads it -- this feels like a unit that can be done independently. However, the main point of not doing all of this through unittest is that loading of course depends on saving. and the latter doesn't fit well with unittest.

Alrighty, unittest is not the right context for this. :)

Reply:
This new torchscript BC test is novel and unique to PyTorch, so I think we should separate it from something as simple as unit test. Following https://github.com/pytorch/pytorch/issues/42258, we have high chance that we need to change the implementation of `AudioMetaData` class, and we need to correctly understand the impact and the test strategy there. So keeping it separate from unit test gives better dev x and make the project more manageable. I would say this is the form of [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns) for project maintanance.